systemtest/src/test/java/io/strimzi/systemtest/kafka/QuotasST.java

```diff
@@ -128,6 +130,83 @@ void testKafkaQuotasPluginIntegration() {
     }
 
+    @ParallelNamespaceTest
+    @Tag(INTERNAL_CLIENTS_USED)
+    @Tag(REGRESSION)
+    void testKafkaQuotasPluginWithBandwidthLimitation() {
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+        final String excludedPrincipal = "User:" + testStorage.getUsername();
+
+        resourceManager.createResourceWithWait(
+            NodePoolsConverter.convertNodePoolsIfNeeded(
+                KafkaNodePoolTemplates.brokerPoolPersistentStorage(testStorage.getNamespaceName(), testStorage.getBrokerPoolName(), testStorage.getClusterName(), 1).build(),
+                KafkaNodePoolTemplates.controllerPoolPersistentStorage(testStorage.getNamespaceName(), testStorage.getControllerPoolName(), testStorage.getClusterName(), 1).build()
+            )
+        );
+        resourceManager.createResourceWithWait(
+            KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), 1, 1)
+                .editSpec()
+                    .editKafka()
+                    .addToConfig("client.quota.callback.static.storage.check-interval", "5")
+                    .withNewQuotasPluginStrimziQuotas()
+                        .addToExcludedPrincipals(excludedPrincipal)
+                        .withProducerByteRate(1000L)    // 1 kB/s
+                        .withConsumerByteRate(1000L)    // 1 kB/s
+                    .endQuotasPluginStrimziQuotas()
+                    .withListeners(
+                        new GenericKafkaListenerBuilder()
+                            .withName(TestConstants.PLAIN_LISTENER_DEFAULT_NAME)
+                            .withPort(9092)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(false)
+                            .build(),
+                        new GenericKafkaListenerBuilder()
+                            .withName("scramsha")
+                            .withPort(9095)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(false)
+                            .withNewKafkaListenerAuthenticationScramSha512Auth()
+                            .endKafkaListenerAuthenticationScramSha512Auth()
+                            .build()
+                    )
+                    .endKafka()
+                .endSpec()
+                .build()
+        );
+        resourceManager.createResourceWithWait(
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), testStorage.getTopicName(), testStorage.getNamespaceName()).build(),
+            KafkaUserTemplates.scramShaUser(testStorage).build()
+        );
+
+        KafkaClients clients = ClientUtils.getInstantPlainClientBuilder(testStorage, KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withMessageCount(100)
+            .withMessage(String.join("", Collections.nCopies(2000, "#")))
+            .withAdditionalConfig("delivery.timeout.ms=10000\nrequest.timeout.ms=10000\n")
```

Comment:
Do you expect failure of this producer? If not, this config is kinda race/flaky incubator. I would keep defaults or at least allow few retries for the messages. Current config says - I will send 1 message and if there will be a problem I will retry it after 10s and fail after 10s so no retry is applied.

Reply:
I see, we can use defaults then. Thanks